src/itemlistformaction.cpp

```diff
@@ -376,6 +377,108 @@ void ItemListFormAction::process_operation(Operation op,
 			v->show_error(_("Error: no item selected!"));
 		}
 	} break;
+	case OP_SAVEALL: {
+		LOG(Level::INFO,
+			"ItemListFormAction: saving all items");
+		if (visible_items.size() == 1) {
+			std::string directory = v->run_dirbrowser("");
+			if (directory.back() != '/') {
+				directory = directory + "/";
+			}
+
+			const std::string filename = v->get_filename_suggestion(visible_items[0].first->title());
+			const std::string fpath = directory + filename;
+
+			struct stat sbuf;
+			if (::stat(fpath.c_str(), &sbuf) != -1) {
+				std::string input_options = _("yn");
+				char c = v->confirm(strprintf::fmt(
+						_("Overwrite %s in %s? "
+						  "(y:Yes n:No)"),
+						filename, directory),
+						input_options);
+				if (!c)
+					break;
+				unsigned int n_options = ((std::string) "yn").length();
+				if (input_options.length() < n_options)
+					break;
+				if (c == input_options.at(0)) {
+					save_article(filename, visible_items[0].first);
+				} else if (c == input_options.at(1)) {
+					save_article(filename, visible_items[0].first);
+				}
+			} else {
+				// get file since it does not exist
+				save_article(filename, visible_items[0].first);
+			}
+
+		} else if (visible_items.size() > 1) {
+			std::string directory = v->run_dirbrowser("");
+			if (directory.back() != '/') {
+				directory = directory + "/";
+			}
+
+			int nfiles_exist = 0;
+			for (const auto &item : visible_items) {
+				const std::string filename = v->get_filename_suggestion(item.first->title());
+				const std::string fpath = directory + filename;
+
+				struct stat sbuf;
+				if (::stat(fpath.c_str(), &sbuf) != -1) {
+					nfiles_exist++;
+				}
+			}
+
+			bool query = true;
```

Comment:
I'd rename that to `overwrite_all` or something else specific. Right now it's not apparent what query is implied (confirmation query? SQL query?), and what `true` and `false` mean (if it's `true`, does it mean we're inside a query, or what?).

Reply:
Just changed in this push.